### DISCUSSION TITLE
Argo Workflow Running Status Is there a real-time Workflow API?

### DISCUSSION THREAD

**Discussion by LingClassmate** [argoproj/argo-workflows#8223] [General]
My application scenario:

I want to push the real-time running status of Argo Workflow task to System A, and system A can synchronize my current workflow status in real time.

**Comment by sarabala1979** [comment]
you can use hooks to get workflow status

**Reply by LingClassmate** [reply]
Do hooks also return task status from Workflow?

**Reply by LingClassmate** [reply]
Hello, I want to return the real-time status of each task in Workflow, can you also use the way you said?

**Reply by sarabala1979** [reply]
yes, there is two levels of hooks, workflow  level and template(step/task)  level

**Reply by LingClassmate** [reply]
I saw it today. Thank you very much for your patience.

**Comment by sarabala1979** [comment]
Yes, there is two type of hook you can configure. Workflow level will trigger workflow status change.
Template level will trigger each step/task status changes
